### ISSUE TITLE
(Enhancement Request) Work off internal cache when not connected to Wifi

### ISSUE THREAD

**Issue by GraphiteWingedBird** [PhilippC/keepass2android#2032]
Long story short, I make the mistake of forgetting to switch my phone to internal cache only when I leave home and if you make an update to an entry (like saving a search URL) it sometimes takes a long time to save if the internet sucks.

It would be cool if there was an option to automatically use the internal cache when not connected to Wifi and then either auto sync the next time wifi is connected, or use the same Sync Database button

**Comment by JusticeWatch** [comment]
I agree, this feature would make the app MUCH easier to use. I have an old phone, and forgetting to toggle the Work On Internal Cache Only box is easy to forget (because the fingerprint prompt comes up on top of it,) and can be costly to my time. 

If the box is unchecked, and I'm out of the house, the app will slow down / freeze with a spinning wheel when trying to look for the source file that doesn't exist on the local network, with no way to cancel the pending transaction. I often have to force close the app to work around this.

If the box is checked, and I'm at home, my cached file may be out of sync with my master file, which can lead to account lockouts, data loss, and most often a waste of time spent locking the database, reopening it, cancelling the fingerprint scan, unchecking the box, and logging in again to force a file synchronization.

The current behavior also poses a small security threat; if I'm connected to some wifi network, and a hacker creates an evil twin mirroring the location of my database, the app could attach to their remote file, sync, and give the attacker a copy of my (encrypted) database.

I would suggest automating the checkbox by adding an optonal check for a specified wifi network and/or certificate before syncing. If the SSID is not connected, use the local cached / backup file by default. This would save a lot of time, checkbox toggling, UI navigation, and headache for the user.

**Comment by WebworkrNet** [comment]
Why limit this feature request to WiFi only and not extend it to an existing Internet connection in general?

**Comment by Spiralpine** [comment]
I fully concur with the opinions expressed in both the initial and second posts. I propose the addition of two checkboxes, ideally positioned directly beneath the "Open file ..." option; one checkbox would allow users to select an online location, directing them to the existing subsequent screen for location choices, while the other would enable work with the Internal Cache. The requested enhancement is much needed.  Please implement.

**Comment by PhilippC** [comment]
> Why limit this feature request to WiFi only and not extend it to an existing Internet connection in general?

for databases stored on a LAN server, it would make sense to even check which Wifi is connected.
